### DISCUSSION TITLE
Consulta creacionPrenda ejercicio queMePongo iteracion 2

### DISCUSSION THREAD

**Discussion by juanmpereira** [dds-jv/foro#110] [Consultas de objetos] [answered]
Buenas, tengo una consulta con respecto al ejercicio de queMePongo iteracion2. 
Al momento de hacer el borrador para crear una prenda, decidi  utilizar la segunda opcion (aquella que en el borrador tengo una prenda en construccion y voy seteando los atributos). Luego el metodo crear prenda en la clase Borrador me quedaria algo asi: 
 
```	
Prenda crearPrenda() {
		this.validar();
		return prendaEnConstruccion = new Prenda();
	}
```

es correcto eso? de serlo, como seria el metodo validacion? deberia tener un constructor en prenda con los datos necesarios para la creacion y llamarlo con '''prenda( prendaEnConstruccion.getColor(), ..) o como lo haria? No me quedo claro esa parte. Espero la respuesta. Saludos

**Comment by gastonprieto** [comment]
@juanmpereira te voy respondiendo por partes

El `new` estaría de mas porque no queres crear una prenda nueva, sino retornar la que ya tenías instanciada dentro del `Borrador`, quedaría algo como:

```Java
Prenda   crearPrenda() {
    ....
    return prendaEnConstruccion;
}
```
> como seria el metodo validacion?

Implementarias todas las validaciones que necesites para la prenda, por ejemplo si decís validar los no nulos ahí en vez del contructor, tendrías algo como:
```
validar()
   notNull(prenda.getTipo())
   ....
```

> deberia tener un constructor en prenda con los datos necesarios para la creacion y llamarlo con '''prenda( prendaEnConstruccion.getColor(), ..) o como lo haria?

Ese código es de instanciación como fuiste por la opción de tener una prenda vacia y luego irle agregando los campos, vos ya lo instanciaste con `new Prenda()` y te lo guardaste como atributo.

**Comment by juanmpereira** [comment]
Entendido, entonces al declarar en la clase Borrador la prenda en construccion deberia hacer: 
Prenda prendaEnConstruccion = new Prenda();
y en prenda tener un constructor vacio. Y luego borrador se encarga de si esa prenda se permite usar si esta completa o no?

**Comment by juanmpereira** [comment]
Y una ultimo duda, como haria para controlar que primero ingrese el tipo y luego el resto de las cosas si lo estoy haciendo a traves de un set?

**Comment by gastonprieto** [comment]
> Entendido, entonces al declarar en la clase Borrador la prenda en construccion deberia hacer:
Prenda prendaEnConstruccion = new Prenda();
y en prenda tener un constructor vacio. Y luego borrador se encarga de si esa prenda se permite usar si esta completa o no?

Correcto

> haria para controlar que primero ingrese el tipo y luego el resto de las cosas si lo estoy haciendo a traves de un set?

Podes asumir que el flujo de entrada se maneja desde la interfaz de usuario, o sino validarlo en el dominio, haciendo que el setter de las otras cosas chequeen que no este en null el `tipo`

**Comment by juanmpereira** [comment, marked as answer]
Perfecto, gracias!
